### PR TITLE
feat: carrier metadata GA

### PR DESCRIPTION
**Pull request by Justintime50**
# Description

- Retrieving carrier metadata is now generally available via `client.carrierMetadata.retrieve`

# Testing

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: src/test/java/com/easypost/CarrierMetadataTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        vcr = new TestUtils.VCR("carrier_metadata", TestUtils.ApiKey.TEST);
+    }
+
+    /**
+     * Test retrieving all carrier metadata.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testRetrieveStatelessRates() throws EasyPostException {
```
Comment (jchen293) [inline src/test/java/com/easypost/CarrierMetadataTest.java:35]: issue: the naming of this test function is incorrect
```suggestion
    public void testRetrieveCarrierMetadata() throws EasyPostException {
```

File: src/test/java/com/easypost/CarrierMetadataTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        assertTrue(carrierMetadata.getCarriers().stream().anyMatch(carrier -> carrier.getName().equals("fedex")));
+    }
+
+    /**
+     * Test retrieving carrier metadata for a specific carrier.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testRetrieveStatelessRatesWithFilter() throws EasyPostException {
```
Comment (jchen293) [inline src/test/java/com/easypost/CarrierMetadataTest.java:50]:
```suggestion
    public void testRetrieveCarrierMetadataWithFilter() throws EasyPostException {
```

**Review by jchen293 — CHANGES_REQUESTED** [review]

File: src/test/java/com/easypost/CarrierMetadataTest.java (hunk 1 above)
Comment (Justintime50) [reply src/test/java/com/easypost/CarrierMetadataTest.java]: That's amazing 😂 This was in the beta one too. Fixed

File: src/test/java/com/easypost/CarrierMetadataTest.java (hunk 2 above)
Comment (Justintime50) [reply src/test/java/com/easypost/CarrierMetadataTest.java]: That's amazing 😂 This was in the beta one too. Fixed

**Review by jchen293 — APPROVED** [review]

File: src/main/java/com/easypost/service/EasyPostClient.java [hunk 3]
```diff
@@ -17,6 +17,7 @@ public class EasyPostClient {
     public final BetaRateService betaRate;
     public final BillingService billing;
     public final CarrierAccountService carrierAccount;
+    public final CarrierMetadataService carrierMetadata;
```
Comment (nwithan8) [inline src/main/java/com/easypost/service/EasyPostClient.java:20]: nit: Should put a deprecated warning on the `public final BetaCarrierMetadataService betaCarrierMetadata` element on line 15.

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**